beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BlockingQueueLogObserver.java

```diff
@@ -143,10 +149,12 @@ public Offset getOffset() {
   private final AtomicLong watermark;
   private final BlockingQueue<Pair<StreamElement, UnifiedContext>> queue;
   private final AtomicBoolean stopped = new AtomicBoolean();
+  private volatile boolean nackAllIncoming = false;
   @Getter @Nullable private UnifiedContext lastWrittenContext;
   @Getter @Nullable private UnifiedContext lastReadContext;
   private long limit;
   private boolean cancelled = false;
+  private transient CountDownLatch cancelledLatch = new CountDownLatch(1);
```

Comment:
unnecessary `cancelled` flag, can be replaced with `cancelledLatch.getCount() > 0`

Reply:
That's technically correct, but the flag is no overhead and makes things a little more explicit. I'd like to keep it there.

Reply:
one think is that the flag is not transient and CountDownLatch is not, so it may not be consistent after serde cycle

Reply:
The latch is not carrying semantic information. It is just a synchronization primitive (which is why it is transient). A cancelled observer remains cancelled even after (de)serialization which is also consistent.